File: samples/perl/lsstatus_inq.pl
#!/usr/bin/perl
use strict;
use warnings;
use JSON;
use LWP::UserAgent;
use HTTP::Request::Common;

# This sample will show the status of all listeners

my $qmgr = shift;
die("Please pass me the name of a queuemanager as argument") 
	unless defined($qmgr);

my $json = JSON->new;

# There's no need to set ListenerName to * because this is the default,
# but we do it here to show how to build the json content.
my %input = ( 
	'ListenerName' => '*',
);
my $content = $json->encode(\%input);    

my $ua = LWP::UserAgent->new;
my $req = POST 'http://localhost:8081/api/lsstatus/inquire/' . $qmgr;
$req->header(
	'Content-Type' => 'application/json',
	'Content-length' => length($content)
);
$req->content($content);

my $res = $ua->request($req);
if ($res->is_success) {
	my $mqweb = $json->decode($res->content());
	if ( exists($mqweb->{error}) ) {
		print "An MQ error occurred while inquiring listener status.\n",
			'Reason Code: ', 
			$mqweb->{error}->{reason}->{code},
			' - ', 
			$mqweb->{error}->{reason}->{desc},
			"\n";
	}
	else {
		foreach my $status(@{$mqweb->{statuses}}) {
			print $status->{ListenerName}->{value};
			print ' : ', $status->{Status}->{display}; 
			print "\n";
		}
	}
}
else {
	die $res->status_line;
}

File: samples/python/lsstatus_inq.py
'''
 Inquire all listener statuses from a queuemanager.
 MQWeb runs on localhost and is listening on port 8081. 
'''
import sys
import json
import httplib
import socket

if len(sys.argv) < 2 :
	print 'Please pass me the name of a queuemanager as argument'
	sys.exit(1)

url = '/api/lsstatus/inquire/' + sys.argv[1]

try:
	conn = httplib.HTTPConnection('localhost', 8081)
	conn.request('GET', url)
	res = conn.getresponse()
	result = json.loads(res.read())

	if 'error' in result:
		print ('Received a WebSphere MQ error: ' +	
			str(result['error']['reason']['code'])
		)
	else:
		if len(result['data']) == 0:
			print "No listener status found"
		else:
			for data in result['data']:
				print ( data['ListenerName']['value'] +
					': ' + data['Status']['text']
				)
except httplib.HTTPException as e:
	print ('An HTTP error occurred while inquiring listener statuses: ' +
		e.errno + e.strerror
	)
except socket.error as e:
	print e.strerror
	print 'Is the MQWeb daemon running?'
